File: qe-selenium-python/Activity_Seventeen.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/selects")
driver.maximize_window()

print(driver.title)

multi_select_element = driver.find_element(By.XPATH, "//select[@multiple]")
multi_select = Select(multi_select_element)

multi_select.select_by_visible_text("HTML")

multi_select.select_by_index(3)
multi_select.select_by_index(4)
multi_select.select_by_index(5)

multi_select.select_by_value("nodejs")

multi_select.deselect_by_index(4)

driver.quit()

File: qe-selenium-java/src/main/java/demos/ActivitySeventeen.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class ActivitySeventeen {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/selects");
        driver.manage().window().maximize();

        System.out.println(driver.getTitle());

        WebElement multiSelectElement = driver.findElement(By.xpath("//select[@multiple]"));
        Select multiSelect = new Select(multiSelectElement);

        multiSelect.selectByVisibleText("HTML");

        multiSelect.selectByIndex(3);
        multiSelect.selectByIndex(4);
        multiSelect.selectByIndex(5);

        multiSelect.selectByValue("nodejs");

        multiSelect.deselectByIndex(4);

        driver.quit();
    }
}
